### PR TITLE
issue #26994 - reduce redundancy in smart bar code handling

### PR DESCRIPTION
**Pull request by gilmoskowitz**
This is a first pass at replacing functions with data to scan smart bar codes. The next step is to drive this mechanism from a database table, allowing extensions to add their own smart bar codes.

### CODE REVIEW

File: guiclient/inputManager.cpp [hunk 1]
```diff
@@ -205,9 +206,9 @@ bool InputManager::eventFilter(QObject *, QEvent *pEvent)
 #ifdef Q_OS_MAC
       case cPrologCtrl:
         // why does character come back as 0?
-	// on an Intel Mac with Qt 4 the key() came back as Key_PageUp
-	// but with PowerPC Mac with Qt 3 the key() came back as 'V'.
-	// Accept either for now.
+        // on an Intel Mac with Qt 4 the key() came back as Key_PageUp
+        // but with PowerPC Mac with Qt 3 the key() came back as 'V'.
```
Comment (davidbeauchamp) [inline guiclient/inputManager.cpp]: Is this still the case?

File: guiclient/inputManagerPrivate.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    int                 _state;
+    int                 _cursor;
+    ScanEvent          *_event;
+    int                 _length1;
+    int                 _length2;
+    int                 _length3;
+    QString             _buffer;
+
+    void dispatchScan(int type);
+    void dispatchWorkOrderOperation();
```
Comment (davidbeauchamp) [inline guiclient/inputManagerPrivate.h]: Are the rest of these dispatch* functions still used? I think I may be missing something, but I only see dispatchScan()   being used, and the rest of the function definitions removed from inputManager.cpp

File: guiclient/inputManager.cpp (hunk 1 above)
Comment (gilmoskowitz) [inline guiclient/inputManager.cpp]: Seems to be, at least with Qt 4.8. Didn't test with Qt 5.

File: guiclient/inputManagerPrivate.h (hunk 2 above)
Comment (gilmoskowitz) [inline guiclient/inputManagerPrivate.h]: Good catch.

**Pull request merged**